clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -269,5 +274,33 @@ data Manifest
     -- ^ Names of all the generated components for the @TopEntity@ (does not
     -- include the names of the components of the @TestBench@ accompanying
     -- the @TopEntity@).
+  } deriving (Show,Read)
+
+-- | Synopsys Design Constraint (SDC) information for a component.
+-- Currently this limited to the names and periods of clocks for create_clock.
+--
+newtype SdcInfo = SdcInfo
+  { sdcClock :: [(Text, VDomainConfiguration)]
   }
-  deriving (Show,Read)
+
+-- | Render an SDC file from an SdcInfo.
+-- The clock periods, waveforms, and targets are all hardcoded.
+--
+pprSDC :: SdcInfo -> Doc ()
+pprSDC = vcat . fmap go . sdcClock
+ where
+  go (i, dom) =
+        -- VDomainConfiguration stores period in ps, SDC expects it in ns.
+    let p        = vPeriod dom `div` 1000
+        name     = braces (pretty i)
+        period   = pretty p
+        waveform = braces ("0" <+> pretty (p `div` 2))
```

Comment:
I might have 5000ps period, i.e 5ns, I then want the the half period to be 2.5ns, so `div 2` won't do. We want to pretty print a `Real` (or `Float`) with 3 decimals of precision.